### PR TITLE
docs: :memo: Create PR template

### PR DESCRIPTION
**Pull request by Nanosplitter**
# Description

This PR adds this template!

# Checklist before marking this PR as ready for review

- [x] All the PR checks are passing (you see the green checkmark)
- [x] I have written comments on the PR where reviewers may have questions

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Description
+
+Describe the changes in this PR and why they are needed.
+
+# Checklist before marking this PR as ready for review
+
+-   [ ] All the PR checks are passing (you see the green checkmark)
+-   [ ] I have written comments on the PR where reviewers may have questions
```
Comment (Nanosplitter) [inline .github/pull_request_template.md:8]: These create empty checkboxes that can be used by the programmers

**Comment by nstringham** [comment]
If your code needs explaining wouldn't it be better to put the comments in source code instead of the PR?

**Comment by Nanosplitter** [comment]
> If your code needs explaining wouldn't it be better to put the comments in source code instead of the PR?

Not if the code needs explaining, but if _why_ you did a thing needs explaining

The code could be super readable, but it might not be known to the reviewer why you made the change, which is what a PR comment is for

**Comment by nstringham** [comment]
> Not if the code needs explaining, but if _why_ you did a thing needs explaining
> 
> The code could be super readable, but it might not be known to the reviewer why you made the change, which is what a PR comment is for
> 
> Like if you delete a block of code, you probably don't want to leave a comment in the source code saying "There was a block of code deleted here because this other feature actually accomplishes all the same logic"
> 
> But that is exactly what you would write in a PR comment

Oh yeah that makes a ton of sense. (Especially if you're squashing)

**Review by nstringham — APPROVED** [review]

**Pull request merged**